### ISSUE TITLE
Adding a module for Publishing Widgets on select pages

### ISSUE THREAD

**Issue by icc365** [BinaryMoon/wp-toolbelt#29]
**Feature Enhancement -** So far things look great, but one very important item to consider is to add a module to the Toolbelt that lets a person publish widgets to select pages/posts/taxonomies, etc. I know Jetpack has Widget Visibility, but it's not 100% functional. It also requires some input by the user in the form of WP conditionals, which is not really something a typical user should use because conditionals are more of a developer thing.

For myself, I've been using the Widget Options plugin, which is pretty good, but it's bloated; but their options to publish widgets is very good. They even have an option to disable widget titles (good if one uses a widget to show a slider or banner image (you don't want a title for it showing). But that is another possible feature and will maybe post that later.

**Comment by BinaryMoon** [comment]
Hi there - thanks for the feedback.

So what you would like is a widget visibility/ widget options style function, but a lot simpler?

I like making things simpler, but I am not sure how to do it for this.

I must admit I have looked at the widget visibility code and it's rather complex, but perhaps there's a way to simplify it. I just need to look at it in a different way! :)

It's on the todo list, and I will have a think and see if I can come up with something.

**Comment by icc365** [comment]
That is the trick, to make things easier and without bloated code. Its unfortunate WordPress does not have this capability built into the core because its a huge benefit to the end-user rather than have a widget show on "all" pages.

I personally would not use the jetpack's widget visibility as a guideline for this, but perhaps look at the "Widget Options" plugin for that particular functionality.

I would say this is something to consider, but for what you have right now, I would say focus on the existing items first, get all the feedback, then once everything is ready to perhaps add this one feature.

If there is an alternative to the Jetpack option, such as your Toolbelt project, I'm all for it. I'm just glad one does not have to register/have an Automattic account for this, as I am sure many will notice that :)

**Comment by BinaryMoon** [comment]
I have a test version of the widget display module

It's quite different to either of the Jetpack or Widget Options ones.

I am trying to keep code as simple as possible since that improves speed, and simplifies development and maintenance. 

* download: https://www.dropbox.com/s/s0qndcwdr9xk2fe/wp-toolbelt.zip?dl=0
* docs: https://github.com/BinaryMoon/wp-toolbelt/wiki/Widget-Display

**Comment by icc365** [comment]
I will try this out later tomorrow as my schedule today is swamped, but looking forward to this version and will try out the widget display module as it relates to this posting. So far though, since I've been trying things out, I have not encountered any bugs or issues.

**Comment by BinaryMoon** [comment]
Hi - it's now part of the plugin! Hopefully it works ok, but if you want any other filters then create a new issue with examples and use cases and I'll do my best to add it.

Thanks for the suggestion!
